### PR TITLE
Enhancements for RDoc

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Methods:
- ::read
- ::readlines
- ::table
- #read

### CODE REVIEW

File: lib/csv.rb [hunk 1]
```diff
@@ -1352,36 +1352,51 @@ def parse_line(line, **options)
...
-    # This method also understands
-    # an additional <tt>:encoding</tt> parameter that you can use to specify the
-    # Encoding of the data in the file to be read. You must provide this unless
-    # your data is in Encoding::default_external(). CSV will use this to determine
-    # how to parse the data. You may provide a second Encoding to have the data
-    # transcoded as it is read. For example,
-    # <tt>encoding: "UTF-32BE:UTF-8"</tt> would read UTF-32BE data from the file
-    # but transcode it to UTF-8 before CSV parses it.
+    # :call-seq:
+    #   read(source, **options ) -> new_csv
```
Comment (kou) [inline lib/csv.rb:1356]:
```suggestion
    #   read(source, **options) -> new_csv
```

Comment (kou) [reply lib/csv.rb:1356]: It doesn't return `new_csv`. 
It returns an array of rows or `CSV::Table`.

Comment (BurdetteLamar) [reply lib/csv.rb]: Ook!  When I typed the call-seq, I didn't know what would be returned.  Then forgot to go back.
Both fixed in next  push.

File: lib/csv.rb [hunk 2]
```diff
@@ -1352,36 +1352,51 @@ def parse_line(line, **options)
...
-    # This method also understands
-    # an additional <tt>:encoding</tt> parameter that you can use to specify the
-    # Encoding of the data in the file to be read. You must provide this unless
-    # your data is in Encoding::default_external(). CSV will use this to determine
-    # how to parse the data. You may provide a second Encoding to have the data
-    # transcoded as it is read. For example,
-    # <tt>encoding: "UTF-32BE:UTF-8"</tt> would read UTF-32BE data from the file
-    # but transcode it to UTF-8 before CSV parses it.
+    # :call-seq:
+    #   read(source, **options) -> array_of_arrays or csv_table
```
Comment (kou) [inline lib/csv.rb:1356]: How about the following?

```rdoc
read(source, **options) -> array_of_arrays
read(source, headers: true, **options) -> csv_table
```

Comment (BurdetteLamar) [reply lib/csv.rb:1356]: Fixed in next push,.

File: lib/csv.rb [hunk 3]
```diff
@@ -1352,36 +1352,52 @@ def parse_line(line, **options)
...
-    # an additional <tt>:encoding</tt> parameter that you can use to specify the
-    # Encoding of the data in the file to be read. You must provide this unless
-    # your data is in Encoding::default_external(). CSV will use this to determine
-    # how to parse the data. You may provide a second Encoding to have the data
-    # transcoded as it is read. For example,
-    # <tt>encoding: "UTF-32BE:UTF-8"</tt> would read UTF-32BE data from the file
-    # but transcode it to UTF-8 before CSV parses it.
+    # :call-seq:
+    #   read(source, **options) -> array_of_arrays
+    #   read(source, header: true, **options) -> csv_table
```
Comment (kou) [inline lib/csv.rb:1357]:
```suggestion
    #   read(source, headers: true, **options) -> csv_table
```

**Comment by kou** [comment]
Thanks.

**Pull request merged**